test

```diff
@@ -0,0 +1,242 @@
+  => Chunk together .in (inputs) and .sol (solution) files
+
+  returns: [["input text", "expected solution"], ...]
+  """
+  dir = '{}/{}/{}/__TESTS__'.format(year, event, problem)
+
+  files = sorted(os.listdir(dir))
+
+  return list(chunks([
+    Path('{}/{}'.format(dir, file)).read_text()
+    for file in files
+  ], 2))
+
+def test(builder, input, expected):
+  """Given a file path, a command, input and expected output,
+  return a 2-uple with a boolean representing if the test passed
+  and the value returned by stderr or stdout"""
+  out, err = builder.run(input)
+
+  solution = bytes.decode(out)
+  error = bytes.decode(err)
+
+  if error:
+    return (False, error)
+
+  if solution == expected:
+    return (True, solution)
+  else:
+    return (False, solution)
+
+def test_all(year, event, problem, language, id):
+  """Test all cases from the given problem"""
+  test_case = cases(year, event, problem)
+
+  file = './{}/{}/{}/{}/solution_{}.{}'.format(year, event, problem, language, id, BUILD_MACHINE[language]['ext'])
+
+  builder = prepare_builder(language, file)
+
+  return test_output([
+    test(builder, input, expected)
+    for input, expected in test_case
+  ])
+
+def prepare_builder(language, file):
+  """Given a language and a file, instance a builder object"""
+  cmdline = BUILD_MACHINE[language]['cmdline']
+  builder = BUILD_MACHINE[language]['builder']
+
+  return builder(cmdline, file)
+
+def test_output(results):
+  """Tell our solver if everything went right"""
+  fails = [
+    (i + 1, output)
+    for i, (passed, output) in enumerate(results)
+    if not passed
+  ]
+
+  if not len(fails):
+    print("Perfect solution!")
+  else:
+    for (i, output) in fails:
+      print(
+        'Wrong response for case {}. Output was {}'
+          .format(i, output[:-1])
+      )
+
+# CLI
+
+parser = OptionParser(
+  usage = 'Usage: ./%prog year problem lang solution_number [--event Main]',
+  version = '1.0'
+)
+
+parser.add_option('-e', '--event',
+                  default="Main")
+
+(options, args) = parser.parse_args()
+
+if len(args) < 4:
+  parser.error('Wrong number of arguments')
+
+if options.event not in ['Main', 'Warming']:
+  parser.error('Wrong kind of event. Must be "Main" or "Warming"')
+
+event = options.event
+year, problem, lang, solution_number = args
+
+if not year.isdigit():
+  parser.error('Year is not a number')
+
+if not solution_number.isdigit():
+  parser.error('Solution number is not a number')
+
+if not lang in BUILD_MACHINE:
+  parser.error(
+    'Language not found.\nAvailable: {}'
+      .format(", ".join(BUILD_MACHINE.keys()))
+  )
+
+possible_file = '{}/{}/{}/{}/solution_{}.{}'.format(year, event, problem, lang, solution_number, BUILD_MACHINE[lang]['ext'])
+
+if not os.path.isfile(os.path.abspath(possible_file)):
+    parser.error('Solution file not found.\nDid you make the file ./{}?'
+                 .format(possible_file))
+
+test_all(int(year), event, problem, lang, int(solution_number))
```

Comment:
You need write the idiomatic on of a call in Python. If you do that, if some weird guy call this shit as a library, this function will be called. Maybe the weird guy maybe do wanna this. To fix that right something like that:

```python

if __name__ == '__main__':
    test_all(int(year), event, problem, lang, int(solution_number))

``` 

So just when the script has the namespace __main__ (called as the main script, like python add.py) this will execute, if is just import, this is ignored. You understand it? Thanks.